src/main/java/Setup.java

```diff
@@ -284,6 +295,61 @@ private static boolean anyDependencyEnabled() {
         return ENABLE_BIGQUERY || ENABLE_AZURE || ENABLE_SNOWFLAKE || ENABLE_REDSHIFT || ENABLE_POSTGRESQL;
     }
 
+    @Override
+    public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+    }
+
+    @Override
+    public void checkServerTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+    }
+
+    @Override
+    public X509Certificate[] getAcceptedIssuers() {
+        return new X509Certificate[0];
+    }
+
+    @Override
+    public List<Proxy> select(URI uri) {
+        return Collections.singletonList(proxy);
+    }
+
+    @Override
+    public void connectFailed(URI uri, SocketAddress sa, IOException ioe) {
+        throw new RuntimeException("Failed to connect to " + uri + " using proxy " + sa);
+    }
+
+    private static final Setup instance = new Setup();
+
+    private static final TrustManager alwaysTrustManager = instance;
+
+    private static final ProxySelector proxySelector = instance;
+
+    private static void setHttpClient() throws NoSuchAlgorithmException, KeyManagementException {
+        setProxy();
+        HttpClient.Builder clientBuilder = HttpClient.newBuilder();
+        clientBuilder.proxy(proxySelector);
+        if (username != null && password != null) {
+            Authenticator authenticator = new Authenticator() {
```

Comment:
doesn't this call create a new . class file ? If so can we make setup inherit Authenticator ?